certification-content-bundle-spec.md

```diff
@@ -0,0 +1,118 @@
+
+**Version 1**
+
+> This is a DRAFT document. We welcome feedback as this format evolves.
+
+## `qwiklabs.yaml` Structure
+
+Here's a sample `qwiklabs.yaml` file with all nested details removed to make it easier to see the general file structure.
+
+```yml
+entity_type: Certification
+schema_version: 1
+
+default_locale: en
+
+title: 
+  locales:
+    en: Virtual Machines in GCP
+
+description:
+  locales:
+    en: Prove what you know about VMs in GCP by doing these courses and exams!
+
+objectives:
+  locales:
+    en:
+      - 1st learning objective
+      - 2nd learning objective
+      - 3rd learning objective
+      - etc...
+
+audience:
+  locales:
+    en: >
+      This certification is intended for people who want to test their knowledge
+      of VMs in GCP.
+
+prerequisites:
+  locales:
+    en: >
+      It is recommended to know a little bit about Unix operating systems before
+      beginning this certification.
+
+tags:
+product_tags:
+role_tags:
+domain_tags:
+
+credits: 80
+
+certificate_award: qwiklabs-accredible
+
+steps: ...
+```
+
+### Certification attributes
+
+attribute          | required | type       | notes
+-------------------| -------- | ---------- | -----------------------------------------
+default_locale     | ✓        | string     | Corresponds to the locale that the certification is authored in. Authoring tools can use this as a hint to notify localizers when content in the default locale is updated. Also, it provides a hint to the learner interface about which locale to display if an instruction/resource is not localized for the learner's current locale.
+schema_version     | ✓        | integer    | Which version of the certification bundle schema you are using
+title              | ✓        | dictionary | A locale dictionary of the certification title, such as "Virtual Machines in GCP"
+description        |          | dictionary | A locale dictionary of the certification description which describes the contents of the certification
+objectives         |          | dictionary | A locale dictionary of objective arrays
+audience           |          | dictionary | A locale dictionary of audiences
+prerequisites      |          | dictionary | A locale dictionary of prerequisites
+tags               |          | array      | Array of strings to be used as hints in searching, etc.
+product_tags       |          | array      | Array of strings from the "Products" column in [this sheet](https://docs.google.com/spreadsheets/d/1hUUch85HBRsRJsgRo9VCg0Pn7ZXi21sl6JU7VOr9LP8)
+role_tags          |          | array      | Array of strings from the "Roles" column in [this sheet](https://docs.google.com/spreadsheets/d/1hUUch85HBRsRJsgRo9VCg0Pn7ZXi21sl6JU7VOr9LP8)
+domain_tags        |          | array      | Array of strings from the "Domain" column in [this sheet](https://docs.google.com/spreadsheets/d/1hUUch85HBRsRJsgRo9VCg0Pn7ZXi21sl6JU7VOr9LP8)
+credits            |          | integer    | Price of the certification
```

Comment:
I'm not sure this belongs here at all: is price an authoring concern? But if it does it should probably be renamed, since `credits` is not a well-known unit.

Reply:
`credits` is used for the Lab price

Reply:
which lab price?

Reply:
We've decided that certifications are offerings, and price is to be defined by the author in the bundle spec.

Reply:
Yeah, I guess these aren't "authors" in the usual sense - they're really just compiling existing content pieces.

Reply:
@shoffing and @davetchen - I'm sorry you need to re-litigate this, but I need to hear the rationale for this. Is there a design doc I can read that summarizes the decision?

Just to be clear about why this is a big deal, this is the first bundle type (other than labs) to include price information. Labs included `price` in credits as an expedient for migration. If we add this to Certifications, then we are officially codifying the pattern that _offering_ concerns are valid for _content_ bundles. That may be the right things to do, but there are a number of consequences.

If we're going to include price (in credits) in content bundle spec, then we need to have plans for 

- Who sets the price? Do they have access to the authoring and publishing workflows?
- What is the value of a credit? In USD? What happens if that exchange rate changes in our system? How do we communicate this relationship to authors?
- Can this price be overridden in certain contexts? (e.g. Catalogs)

To be clear, I firmly believe that information in the content bundle spec can _inform_ pricing and offering configurations. 

The simplest implementation could be an MSRP in USD that is displayed when configuring catalog prices (in credits). This MSRP would have no programmatic effect on content purchasing.

A more complicated system for Labs would be us analyze a number of (related) factors, such as average resource usage and abuse risk. The result would inform (at offer time) the minimum cost in credits for lab in any public catalog. We would be free to play with the value of credits and experiment with what those pricing thresholds are to maximize legitimate usage without increased risk.

But given:
(1) we're pretty far away from implementing that system, and
(2) we have high-touch relationships with everyone who creates self-paced lab content.

Therefore, pushing the decision of cost-in-credits as a heuristic onto authors is currently worth the drawbacks to our system overall.

I don't want to take that hit with a new content type that does not significantly benefit from tying itself to credits at authoring/publishing time.

Reply:
I am going to stick my nose into this. I don't fully understand the issue here so I am going to ask a bunch of questions:
1- If price is not an authoring concern, then where is it supposed to be set in the Qwiklabs eco-system?
2- If price is a publishing concern, would that change where it should be set?
3- Should one be allowed to change price in a production deployment? Is that what's happening with the Catalog override?
4- How do we set the price for exams if not in the content-bundle? Especially in the short-term?
5- Who is supposed to set the price if it is not the creators?
6- I don't understand why the QL Credit to USD exchange rate has anything to do with this. Can I ask for more detail on why that is a concern?

Reply:
Might be worth getting the group (Steve, Dave, Enis, Josh, me, ...?) together for a 30-minute chat.